pkgs/games/unvanquished/default.nix

```diff
@@ -0,0 +1,139 @@
+        echo ${gcc.cc.lib}/lib/libstdc++.so.6 > $out/etc/ld-nix.so.preload
+      '';
+    };
+
+    fhsEnv = buildFHSUserEnv {
+      name = "unvanquished-fhs-wrapper";
+      targetPkgs = pkgs: [ gcc.cc.lib preload ];
+    };
+
+    unvanquished-assets = stdenv.mkDerivation {
+      name = "unvanquished-assets";
+      inherit version;
+      src = fetchurl {
+        # master as of 2021-04-09: the 0.51.2 script isn't as great
+        url = "https://raw.githubusercontent.com/Unvanquished/Unvanquished/2783e77acc2e41fc1f717124394ed851f1b612e6/download-paks";
+        sha256 = "1zbsi0yzfvgassg6dqks66qpjqa4yxi2ymis2qh8aj9dvkgqk6im";
+      };
+      outputHash = "sha256:15vndprms0vi5rbvf67hqxwgs3xhlcrxbzn4z3kz9d54g0z1zbzh";
+      outputHashMode = "recursive";
+      nativeBuildInputs = [ aria2 cacert ];
+      meta = {
+        license = lib.licenses.cc-by-sa-25;
+        description = "Assets for the Unvanquished game";
+      };
+      buildCommand = "bash $src .";
+      installPhase = "cp ./* $out/";
+    };
+
+# this really is the daemon game engine, the game itself is in the assets
+in stdenv.mkDerivation rec {
+  name = "unvanquished-${version}";
+  inherit version;
+  inherit binary-deps-version;
+  src = fetchgit {
+    url = "https://github.com/Unvanquished/Unvanquished.git";
+    rev = "v${version}";
+    sha256 = "1x943sxbgd41dhhc4j43y8vdwj74g9rbxyz1d7an82zbjsp34lak";
+  };
+  cmakeFlags = "-DBUILD_CGAME=NO -DBUILD_SGAME=NO";
+    #USE_HARDENING=TRUE
+    #USE_LTO=TRUE
+  preConfigure = ''
+    mkdir daemon/external_deps/linux64-${binary-deps-version}/
+    cp -r ${unvanquished-binary-deps}/* daemon/external_deps/linux64-${binary-deps-version}/
+    chmod +w -R daemon/external_deps/linux64-${binary-deps-version}/
+  '';
+  nativeBuildInputs = [ cmake unvanquished-binary-deps ];
+  buildInputs = [ gmp libGL zlib ncurses geoip lua5
+                  nettle curl SDL2 freetype glew openal
+                  libopus opusfile libogg libvorbis libjpeg libwebp libpng
+                  libtheora # TODO: drop this one when 0.52 is released
+  ];
+  installPhase = ''
+    # install the binaries
+    for f in daemon daemon-tty daemonded nacl_loader nacl_helper_bootstrap; do
+      install -Dm0755 -t $out/lib/ $f
+    done
+    install -Dm0644 -t $out/lib/ irt_core-x86_64.nexe
+
+    # and a wrapper for them
+    mkdir $out/bin/
+    cat > $out/bin/daemon-unwrapped <<EOT
+    #!/bin/sh
+    exec $out/lib/daemon -pakpath ${unvanquished-assets} "\$@"
+    EOT
+    chmod +x $out/bin/daemon-unwrapped
+    cat > $out/bin/daemon <<EOT
+    #!/bin/sh
+    exec ${fhsEnv}/bin/unvanquished-fhs-wrapper $out/bin/daemon-unwrapped "\$@"
+    EOT
+    chmod +x $out/bin/daemon
+
+    # and the other desktop thingies
+    install -Dm0644 -t $out/share/applications/ ${makeDesktopItem {
+      name = "Unvanquished";
+      desktopName = "unvanquished";
+      exec = "daemon";
+      icon = "unvanquished";
+      categories = "Game;";
+    }}/share/applications/*
```

Comment:
Please use copyDesktopItems instead.

Reply:
I tried searching for some copyDesktopItems documentation, but could not find any. So I tried to have a look at examples, but unfortunately I can't get it to work.

Reply:
Does https://github.com/NixOS/nixpkgs/blob/7248f89da2dab8bca3a9f4f11d0407d0c8e25e89/pkgs/applications/misc/dbeaver/default.nix#L74 help you?

Reply:
I think I was missing the explicit runHook calls